v2/internal/attacktechniques/aws/credential-access/secretsmanager-batch-retrieve-secrets/main.go

```diff
@@ -0,0 +1,148 @@
+		FriendlyName: "Retrieve a High Number of Secrets Manager secrets (Batch)",
+		Description: `
+Retrieves a high number of Secrets Manager secrets by batch, through <code>secretsmanager:BatchGetSecretValue</code> (released Novemeber 2023). 
+An attacker may attempt to retrieve a high number of secrets by batch, to avoid detection and generate fewer calls. Note that the batch size is limited to 20 secrets.
+
+
+Warm-up: 
+
+- Create multiple secrets in Secrets Manager.
+
+Detonation: 
+
+- Dump all secrets by batch of ` + strconv.Itoa(BatchSize) + `, using <code>secretsmanager:BatchGetSecretValue</code>.
+
+References:
+
+- https://aws.amazon.com/blogs/security/how-to-use-the-batchgetsecretsvalue-api-to-improve-your-client-side-applications-with-aws-secrets-manager/
+`,
+		Detection: `
+Identify principals that attempt to retrieve secrets by batch, through CloudTrail's <code>BatchGetSecretValue</code> event. Sample event:
+
+` + code + `json
+{
+  "eventSource": "secretsmanager.amazonaws.com",
+  "eventName": "BatchGetSecretValue",
+  "requestParameters": {
+    "filters": [
+      {
+        "key": "tag-key",
+        "values": [
+          "StratusRedTeam"
+        ]
+      }
+    ]
+  },
+  "responseElements": null,
+  "readOnly": true,
+  "eventType": "AwsApiCall",
+  "managementEvent": true,
+  "recipientAccountId": "012345678901"
+}
+` + code + `
+
+Although <code>BatchGetSecretValue</code> requires a list of secret IDs or a filter, an attacker may use a catch-all filter to retrieve all secrets by batch:
+
+` + code + `json hl_lines="6-11"
+{
+  "eventSource": "secretsmanager.amazonaws.com",
+  "eventName": "BatchGetSecretValue",
+  "requestParameters": {
+    "filters": [
+      {
+        "key": "tag-key",
+        "values": [
+          "!tagKeyThatWillNeverExist"
+        ]
+      }
+    ]
+  },
+  "responseElements": null,
+  "readOnly": true,
+  "eventType": "AwsApiCall",
+  "managementEvent": true,
+  "recipientAccountId": "012345678901"
+}
+` + code + `
+
+The following may be use to tune the detection, or validate findings:
+
+- Principals who do not usually call GetBatchSecretValue
+- Attempts to call GetBatchSecretValue resulting in access denied errors
+- Principals calling GetBatchSecretValue in several regions in a short period of time`,
+		Platform:                   stratus.AWS,
+		IsIdempotent:               true,
+		MitreAttackTactics:         []mitreattack.Tactic{mitreattack.CredentialAccess},
+		PrerequisitesTerraformCode: tf,
+		Detonate:                   detonate,
+	})
+}
+
+func detonate(_ map[string]string, providers stratus.CloudProviders) error {
+	secretsManagerClient := secretsmanager.NewFromConfig(providers.AWS().GetConnection())
+	filters := []types.Filter{
+		{Key: types.FilterNameStringTypeTagKey, Values: []string{"StratusRedTeam"}},
+	}
+
+	_, err := secretsManagerClient.ListSecrets(context.Background(), &secretsmanager.ListSecretsInput{
```

Comment:
This is not needed now you can call the getSecretValue directly, no?

Reply:
good catch, copy paste leftover... thanks!